src/grandorgue/model/pipe-config/GOPipeConfig.h

```diff
@@ -128,6 +130,11 @@ class GOPipeConfig {
     SetPitchMember(cents, m_AutoTuningCorrection);
   }
 
+  int GetToneBalanceValue() const { return m_ToneBalanceValue; }
+  void SetToneBalanceValue(int value) {
+    SetToneBalanceMember(value, m_ToneBalanceValue);
+  }
```

Comment:
```suggestion
  int8_t GetToneBalanceValue() const { return m_ToneBalanceValue; }
  void SetToneBalanceValue(int8_t value) {
    SetSmallMember(std::clamp(value, -100, 100), m_ToneBalanceValue, &GOPipeUpdateCallback::UpdateToneBalance);
  }
```

Reply:
std::clamp is strictly a C++17 feature without any fallback on for instance Ubuntu 20.04. Therefore I think we should keep the manual range checking until the lowest version we build on fully supports the feature.

Reply:
Yes, it a C++17 feature but it is present in Ubuntu 20.04. Since #1845 building GO requires c++17. ``std::clamp`` is already used in GrandOrgue, for example, [here](https://github.com/GrandOrgue/grandorgue/blob/71e02e6db8d0811705faa073c37408e501953aff/src/grandorgue/wxcontrols/GOGrid.cpp#L60)

Reply:
Very strange that I then couldn't get GO to compile with it when I tested! And I did include <algorithm>.

Reply:
What error message did you receive when you was trying to compile on ubuntu 20.04?

Reply:
Sorry, can't exactly remember. I think it was something about no matching function to call, so it might actually have been related to the SetSmallMember function more than the clamp itself. I admit that I didn't try moving it out from the function while testing.

Reply:
I made using clamp based on your code. Everething is compiling under ubuntu 20.04: https://github.com/oleg68/GrandOrgue-official/actions/runs/9045413075

I made [a PR to your branch](https://github.com/larspalo/grandorgue/pull/1141). Could you take a look?

Reply:
Ok, so the problem was that the two other parameters didn't default to int8_t by the value given alone, but perhaps int instead, and then the function wouldn't work and it needs a cast to int8_t to work explicitly as a template, right?

Reply:
Yes, it is. `std::clamp(int8_t, int, int)` does not exist.